src/pollen.py

```diff
@@ -0,0 +1,99 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Pollen charm business logic."""
+
+import glob
+import subprocess
+from pathlib import Path
+
+from charms.operator_libs_linux.v0 import apt
+from charms.operator_libs_linux.v1 import systemd
+from charms.operator_libs_linux.v2 import snap
```

Comment:
It seems that we're using different versions of the same library for different purposes? Perhaps worth an explanation as a comment?

Reply:
Those are just the versions of those libraries that the relevant commands are available in.

Reply:
Each lib is available only on that version, I'll add a comment anyway :+1: